### PR TITLE
Kaleab.about-page

### PR DESCRIPTION
**Pull request by kkstarkk**
In this branch I have implemented the About Us Page. I have divided the page into 4 sections: Hero Section, Problems Section, 
Social Projects Section and Sessions Section.
**Hero Section:**
Figma
image
Implementation:
image

**Problems Section:**
Figma:
image
Implementation:
image

**Social Projects Section:**
Figma:
image

Implementation:
image

**Sessions Section:**
Figma:
image

Implementation:
image

### CODE REVIEW

File: Web/starter_project_barbies/components/about/HeroSection.tsx
Comment (Adona12) [inline Web/starter_project_barbies/components/about/HeroSection.tsx:3]: Can you use syntax like React.FC to define a component? It is better to make it clear that we are looking at a component

File: Web/starter_project_barbies/public/about/africa.svg
Comment (Adona12) [inline Web/starter_project_barbies/public/about/africa.svg:1]: Let's have a clear folder structure. Images should be inside the img folder then we can have a subfolder inside it

File: Web/starter_project_barbies/components/about/HeroSection.tsx
Comment (Adona12) [inline Web/starter_project_barbies/components/about/HeroSection.tsx:8]: Let's define colors globally(tailwind.config.js) so others can also use it.

File: Web/starter_project_barbies/public/about/amicopuzzle.png
Comment (Adona12) [inline Web/starter_project_barbies/public/about/amicopuzzle.png:1]: Use kebab case for everything except the components. You should separate the words with a dash

File: Web/starter_project_barbies/components/about/ProjectCard.tsx
Comment (Adona12) [inline Web/starter_project_barbies/components/about/ProjectCard.tsx:4]: Let's define props here above the component so it's more clear. I think it would be best to do it for all the components whose props are define on types folder

File: Web/starter_project_barbies/public/about/approach.svg
Comment (Adona12) [inline Web/starter_project_barbies/public/about/approach.svg:1]: I would suggest having subgroups for the images, you can put session related images in a session folder, same thing for projects and you can be creative and create folder for the others. The main thing is to divide them rather than having all this image in just one folder.

File: Web/starter_project_barbies/data/about.ts
Comment (YiseBoge) [inline Web/starter_project_barbies/data/about.ts:1]: can we have this data be from a json file?
i woule personally prefer it as it lets us treat it similar to what we would find from backend

Comment (YiseBoge) [reply Web/starter_project_barbies/data/about.ts:1]: it is fine to have multiple json files if you need

File: Web/starter_project_barbies/data/about.ts
Comment (YiseBoge) [inline Web/starter_project_barbies/data/about.ts:5]: none of the links should have the `/about` in them
you should decide that when you are rendering the page because it is common for all)
it also gives the flexibility to change that folder without having to modify all these links

File: Web/starter_project_barbies/data/about.ts
Comment (YiseBoge) [inline Web/starter_project_barbies/data/about.ts:34]: [nit] there is no need for `//` (it appears multiple places)

File: Web/starter_project_barbies/data/about.ts
Comment (YiseBoge) [inline Web/starter_project_barbies/data/about.ts:69]: if the ProblemItem type is the same one we use for SolutionItem too it might be better to rename it to be inclusive

Comment (YiseBoge) [reply Web/starter_project_barbies/data/about.ts:69]: alternatively, you can define a SolutionItem as 
`type SolutionItem = ProblemItem`
in your type definitions to have a copy of it that acts the same way but it has a clearer name

File: Web/starter_project_barbies/components/about/SessionList.tsx
Comment (YiseBoge) [inline Web/starter_project_barbies/components/about/SessionList.tsx:1]: I think it might be fine to define the SessionCard component inside this file 
and I would call this component `SessionsList` (Sessions plural)
this is because SessionCard is only used in this component and it is better to define it here and not have to export it
They are also both small components so it won't be too much for one file

File: Web/starter_project_barbies/components/about/SessionCard.tsx
Comment (YiseBoge) [inline Web/starter_project_barbies/components/about/SessionCard.tsx:1]: the prop type definition should follow a pattern of `[ComponentName]Props` so this should be `SessionCardProps`

File: Web/starter_project_barbies/components/about/ProblemItemCard.tsx
Comment (YiseBoge) [inline Web/starter_project_barbies/components/about/ProblemItemCard.tsx:1]: this can be defined inside the ProblemSection file as it is only used in there and it is better not to have to export it

File: Web/starter_project_barbies/components/about/ProblemSection.tsx
Comment (YiseBoge) [inline Web/starter_project_barbies/components/about/ProblemSection.tsx:1]: [nit] I would rename this to `ProblemsSection` because it holds multiple problems

File: Web/starter_project_barbies/components/about/ProblemItemCard.tsx
Comment (YiseBoge) [reply Web/starter_project_barbies/components/about/ProblemItemCard.tsx:1]: and it is small enough to not be a problem too

File: Web/starter_project_barbies/components/about/ProblemSection.tsx
Comment (YiseBoge) [inline Web/starter_project_barbies/components/about/ProblemSection.tsx:1]: variable names should follow `camelCase` format (these are not class definitions)

File: Web/starter_project_barbies/components/about/HeroSection.tsx
Comment (YiseBoge) [reply Web/starter_project_barbies/components/about/HeroSection.tsx:8]: i think you should be using the primary color here as well (make sure the primary is defined appropriately too)
because the current color feels a bit off in the UI

File: Web/starter_project_barbies/components/about/HeroSection.tsx
Comment (YiseBoge) [inline Web/starter_project_barbies/components/about/HeroSection.tsx:40]: [very-nit] `relative  flex-1` has an extra space in between

File: Web/starter_project_barbies/components/about/HeroSection.tsx
Comment (YiseBoge) [inline Web/starter_project_barbies/components/about/HeroSection.tsx:29]: [very-nit] `relative flex-1 ` has an extra space in the end

File: Web/starter_project_barbies/components/about/ProjectCard.tsx
Comment (YiseBoge) [inline Web/starter_project_barbies/components/about/ProjectCard.tsx:4]: instead of `isEven`, i would rather take in `shouldReverseFlex` or sth
basically something that describes the UI because the only reason we need the evenness is to check if it should reverse (we should not care about why it should reverse in here)

Comment (YiseBoge) [reply Web/starter_project_barbies/components/about/ProjectCard.tsx:4]: in this case of course all you'd need to do is rename the variable
but i just wanted to explain the concept

File: Web/starter_project_barbies/components/about/ProjectCard.tsx
Comment (YiseBoge) [inline Web/starter_project_barbies/components/about/ProjectCard.tsx:36]: a lot of these images can be replaced with icons 
better to do that if possible

File: Web/starter_project_barbies/components/about/ProjectCard.tsx
Comment (YiseBoge) [inline Web/starter_project_barbies/components/about/ProjectCard.tsx:16]: [very-nit] i think you don't need to condense the alt too much, you can just say Hakim Hub or sth

File: Web/starter_project_barbies/components/about/ProjectCard.tsx
Comment (YiseBoge) [inline Web/starter_project_barbies/components/about/ProjectCard.tsx:7]: [nit] can you try this format:

```
className={{
  flex,
  flexWrap,
  justifyAround,
  flexRowReverse: isEven
}}
// not entirely sure if the keys should be in camelCase or kebab-case with quotes
```

it is an alternative approach that might be cleaner

File: Web/starter_project_barbies/components/about/ProjectCard.tsx
Comment (YiseBoge) [inline Web/starter_project_barbies/components/about/ProjectCard.tsx:1]: The project card could be defined in the same file as the project list as that is the only place it is used
it might be better to not have to export it

File: Web/starter_project_barbies/components/about/ProjectList.tsx
Comment (YiseBoge) [inline Web/starter_project_barbies/components/about/ProjectList.tsx:1]: variables should be in camelCase

File: Web/starter_project_barbies/components/about/ProjectList.tsx
Comment (YiseBoge) [inline Web/starter_project_barbies/components/about/ProjectList.tsx:1]: i would rename this to `ProjectsList` (it holds multiple projects)

File: Web/starter_project_barbies/public/about/hakimhub.png
Comment (YiseBoge) [inline Web/starter_project_barbies/public/about/hakimhub.png:1]: [nit] could be named `hakim-hub`

**Review by YiseBoge — CHANGES_REQUESTED** [review]
The implementation seems nice overall
i have left some comments (most of them are minor)
some are related to having too many files (be a bit more careful with their naming standards too)

**Review by sam5235 — APPROVED** [review]

**Comment by YiseBoge** [comment]
let's please rename the PR title to follow the conventional commit rules
[nit] `kaleab.about-page` is a better name for the branch for next time

File: Web/starter_project_barbies/components/about/ProjectsList.tsx
Comment (YiseBoge) [inline Web/starter_project_barbies/components/about/ProjectsList.tsx:53]: [nit] please use React.FC syntax to define components

File: Web/starter_project_barbies/data/about.tsx
Comment (YiseBoge) [reply Web/starter_project_barbies/data/about.tsx:1]: it might be fine to do this in another PR if you want to, but it will be necessary for the next task

**Review by YiseBoge — CHANGES_REQUESTED** [review]
i have left some nits that i think should take you short time to finish
and please lint the project before pushing and merge any conflicts if they arise

File: Web/starter_project_barbies/components/about/ProblemsSection.tsx
Comment (YiseBoge) [inline Web/starter_project_barbies/components/about/ProblemsSection.tsx:8]: [nit] I would rename `Item` to `problemItem` as that is what it seems to be and you should stay consistent with the camelCase

File: Web/starter_project_barbies/components/about/ProblemsSection.tsx
Comment (YiseBoge) [inline Web/starter_project_barbies/components/about/ProblemsSection.tsx:52]: again, lets stick to camelCase for the variable name, and let's make sure we rename it appropriately too
Item here is being used to represent both solutionItem and problemItem above in the same file. This can be confusing

File: Web/starter_project_barbies/components/about/ProjectsList.tsx
Comment (YiseBoge) [inline Web/starter_project_barbies/components/about/ProjectsList.tsx:16]: [very-nit] one alternative way to do this if you want is
```
className={{
  flex,
  flexWrap,
  justifyAround,
  flexRowReverse: reverseFlex
}}
```
(not entirely sure if the class names need to be camelCase or 'kebab-case' as strings)

No need to change now, just wanted to show you an alternative way if you want. this way could be useful when you have class names that will be needed only upon a condition (the ones that should always be there will just be listed as if their condition is true). and it would be cleaner expecially if there are multiple of them

File: Web/starter_project_barbies/pages/about/index.tsx
Comment (YiseBoge) [inline Web/starter_project_barbies/pages/about/index.tsx:1]: [nit] i think you can use `@/components/...` for your paths (this might apply elsewhere too)

**Review by YiseBoge — CHANGES_REQUESTED** [review]
This looks really nice
i have left some small nits on it tho, should be 5 minute fix

File: Web/starter_project_barbies/components/about/ProblemsSection.tsx
Comment (kkstarkk) [reply Web/starter_project_barbies/components/about/ProblemsSection.tsx:8]: i named it Item because the prop could be problemItem or solutionItem. shall i rename the component to ItemCard instead??

Comment (kkstarkk) [reply Web/starter_project_barbies/components/about/ProblemsSection.tsx:8]: i named it Item because the prop could be problemItem or solutionItem. shall i rename the component to ItemCard instead??

File: Web/starter_project_barbies/components/about/ProjectsList.tsx
Comment (YiseBoge) [inline Web/starter_project_barbies/components/about/ProjectsList.tsx:37]: [nit] this is fine for now but generally it is better to not have the negative in the conditional, I'd rather you swap the justify start and justify end instead of adding a not on the reverseFlex
I think you've done that other places too

**Review by YiseBoge — APPROVED** [review]
This looks good
Well done

**Pull request merged**
